Feature/vplay 9633

hthakkar-synamedia opened the pull request:
Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                         A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.
Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API
Test Procedure: Refer to the Ticket.
Risks: Low

hthakkar-synamedia committed:
First Draft

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=282) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=282) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=282) it.</sub>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
7c86cd286b28a147fd401d4fa2f3eeab
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"processor" is copied in call to copy assignment for class "std::shared_ptr<IsoBmffProcessor>", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""processor"")" instead of "processor".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:4033](undefined//blob/undefined/streamabstraction.cpp#L4033)

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low
Signed-off-by: hthakkar-synamedia <115170760+hthakkar-synamedia@users.noreply.github.com>

hthakkar-synamedia committed:
Merge branch 'dev_sprint_25_2' of github.com:rdkcentral/aamp into feature/VPLAY-9633

VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

Signed-off-by: hthakkar-synamedia <115170760+hthakkar-synamedia@users.noreply.github.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9633

hthakkar-synamedia commented on test/utests/tests/MediaTrackTests/MediaTrackTests.cpp:
Done

hthakkar-synamedia commented on test/utests/tests/StreamAbstractionAAMP/FunctionalTests.cpp:
```diff
@@ -191,3 +206,30 @@ TEST_F(StreamAbstractionAAMP_Test, WaitFor_VideoTrackCatchup_discontinuity)
+	EXPECT_CALL(*mStreamAbstractionAAMP, clearFirstPTS());
+	EXPECT_CALL(*mStreamAbstractionAAMP->mMockAudioTrack, ResetTrickModePtsRestamping()).Times(1);
+	EXPECT_CALL(*mStreamAbstractionAAMP->mMockVideoTrack, ResetTrickModePtsRestamping()).Times(1);
+	EXPECT_CALL(*mMockMediaProcessor, setRate(_, _)).Times(1);
```
Done

hthakkar-synamedia commented on test/utests/tests/fragmentcollector_mpd/fragmentcollector_mpd1.cpp:
```diff
@@ -204,7 +205,7 @@ class MpdTests : public ::testing::Test
 		mStreamAbstractionAAMP_MPD = new TestableStreamAbstractionAAMP_MPD(mPrivateInstanceAAMP);
 		EXPECT_CALL(*g_mockPrivateInstanceAAMP, DownloadsAreEnabled()).WillRepeatedly(Return(true));
-
+		EXPECT_CALL(*g_mockPrivateInstanceAAMP, GetLLDashChunkMode()).Times(AnyNumber());
```
Done

hthakkar-synamedia committed:
Merge branch 'dev_sprint_25_2' of github.com:rdkcentral/aamp into feature/VPLAY-9633

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

Signed-off-by: hthakkar-synamedia <115170760+hthakkar-synamedia@users.noreply.github.com>

hthakkar-synamedia commented on test/utests/tests/StreamAbstractionAAMP_MPD/AudioOnlyTests.cpp:
```diff
@@ -278,7 +278,8 @@ class AudioOnlyTests : public ::testing::Test
 			.Times(AnyNumber())
 			.WillRepeatedly(Return(eSTATE_PREPARING));
-
+		EXPECT_CALL(*g_mockPrivateInstanceAAMP, GetLLDashChunkMode()).Times(AnyNumber());
```
Done

hthakkar-synamedia commented on test/utests/tests/CacheFragmentTests/CacheFragmentTests.cpp:
```diff
@@ -39,6 +39,7 @@ using ::testing::Return;
 using ::testing::StrictMock;
 using ::testing::SetArgReferee;
 using ::testing::AtLeast;
+using ::testing::AnyNumber;
```
Removed

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position
                     A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

Signed-off-by: hthakkar-synamedia <115170760+hthakkar-synamedia@users.noreply.github.com>

hthakkar-synamedia committed:
Merge branch 'dev_sprint_25_2' of github.com:rdkcentral/aamp into feature/VPLAY-9633

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f6eeb33ad2f6f3c3baf50a4663f07788
PRESENT
-->

# Coverity Issue - Data race condition
A wait is performed without a loop. If there is a spurious wakeup, the condition may not be satisfied.

_Medium Impact, CWE-none_
_BAD_CHECK_OF_WAIT_COND_

## How to fix
 Check the wait condition in a loop, with the lock held. The lock must not be released between the condition and the wait.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:7467](undefined//blob/undefined/priv_aamp.cpp#L7467)

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9633

Vinish100 reviewed (approved):
Looks good 👍

hthakkar-synamedia committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9633

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fa66964f0e9f006f23d564ed80411b71
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->cachingCompleted" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.cachingCompleted" is written to with "MediaTrack.mutex" held 3 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:3404](undefined//blob/undefined/streamabstraction.cpp#L3404)

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9633
